src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1674,18 +1674,26 @@ void TorrentImpl::handleTrackerWarningAlert(const lt::tracker_warning_alert *p)
 
 void TorrentImpl::handleTrackerErrorAlert(const lt::tracker_error_alert *p)
 {
-    const QString trackerUrl = p->tracker_url();
-
     // Starting with libtorrent 1.2.x each tracker has multiple local endpoints from which
     // an announce is attempted. Some endpoints might succeed while others might fail.
     // Emit the signal only if all endpoints have failed.
-    const QVector<TrackerEntry> trackerList = trackers();
-    const auto iter = std::find_if(trackerList.cbegin(), trackerList.cend(), [&trackerUrl](const TrackerEntry &entry)
+    for (const lt::announce_entry &tracker : m_nativeHandle.trackers())
     {
-        return (entry.url == trackerUrl);
-    });
-    if ((iter != trackerList.cend()) && (iter->status == TrackerEntry::NotWorking))
-        m_session->handleTorrentTrackerError(this, trackerUrl);
+        if (p->tracker_url() != tracker.url)
+            continue;
+
+        const QString trackerURL = QString::fromStdString(tracker.url);
+
+#ifdef QBT_USES_LIBTORRENT2
+        TrackerEntry entry = fromNativeAnnouncerEntry(tracker, m_nativeHandle.info_hashes(), m_trackerPeerCounts[trackerURL]);
+#else
+        TrackerEntry entry = fromNativeAnnouncerEntry(tracker, m_trackerPeerCounts[trackerURL]);
+#endif
+
+        if (entry.status == TrackerEntry::NotWorking)
+            m_session->handleTorrentTrackerError(this, trackerURL);
+        break;
```

Comment:
```suggestion
        if (p->tracker_url() == tracker.url)
        {
            const QString trackerURL = QString::fromStdString(tracker.url);
#ifdef QBT_USES_LIBTORRENT2
            TrackerEntry entry = fromNativeAnnouncerEntry(tracker, m_nativeHandle.info_hashes(), m_trackerPeerCounts[trackerURL]);
#else
            TrackerEntry entry = fromNativeAnnouncerEntry(tracker, m_trackerPeerCounts[trackerURL]);
#endif

            if (entry.status == TrackerEntry::NotWorking)
                m_session->handleTorrentTrackerError(this, trackerURL);
        
            break;
        }
```

Reply:
I'm pretty sure @Chocobo1 will object to extra indentation level

Reply:
> I'm pretty sure @Chocobo1 will object to extra indentation level

I'm afraid I won't agree with him in this case. Two loop interrupt instructions associated with one condition is too much. Besides, the last `break` looks really confusing.

Reply:
Or just use `std::find_if` to find tracker entry with given url and then place another code.

Reply:
>Or just use std::find_if to find tracker entry with given url and then place another code.

This.